apiserver/facades/client/application/application_get.go

```diff
@@ -129,6 +129,7 @@ func (api *APIBase) getCharmID(ctx context.Context, charmURL string) (corecharm.
 	charmID, err := api.applicationService.GetCharmID(ctx, applicationcharm.GetCharmArgs{
 		Name:     curl.Name,
 		Revision: ptr(curl.Revision),
+		Source:   applicationcharm.CharmSource(curl.Schema),
```

Comment:
We should try to start adding test coverage to this facade package.

Reply:
Why not `ParseCharmSource`?

Reply:
Oups, I committed to quickly, fixing it.

Reply:
> We should try to start adding test coverage to this facade package.

I wasn't sure how much are we planning to bulldozer of these facades, I can start adding some now.

Reply:
@jack-w-shaw added `ParseCharmSource`. @SimonRichardson I added 2 basic tests for the deploy (and with a bad source) code paths. The idea is to start adding more and more as soon as we continue removing calls to state.